symfony/plugins/orangehrmAttendancePlugin/Api/AttendanceConfigurationAPI.php

```diff
@@ -0,0 +1,541 @@
+                    WorkflowStateMachine::FLOW_ATTENDANCE,
+                    AttendanceRecord::STATE_PUNCHED_IN,
+                    self::ESS_USER,
+                    WorkflowStateMachine::ATTENDANCE_ACTION_DELETE,
+                    AttendanceRecord::STATE_NA
+                );
+            }
+
+            $isPunchOutRecordDeletable = $this->getAttendanceService()->getAttendanceDao()->getSavedConfiguration(
+                WorkflowStateMachine::FLOW_ATTENDANCE,
+                AttendanceRecord::STATE_PUNCHED_OUT,
+                self::ESS_USER,
+                WorkflowStateMachine::ATTENDANCE_ACTION_DELETE,
+                AttendanceRecord::STATE_NA
+            );
+
+            if (!$isPunchOutRecordDeletable) {
+                $this->saveConfiguration(
+                    WorkflowStateMachine::FLOW_ATTENDANCE,
+                    AttendanceRecord::STATE_PUNCHED_OUT,
+                    self::ESS_USER,
+                    WorkflowStateMachine::ATTENDANCE_ACTION_DELETE,
+                    AttendanceRecord::STATE_NA
+                );
+            }
+        } else {
+            $this->getAccessFlowStateMachineService()->deleteWorkflowStateMachineRecord(
+                WorkflowStateMachine::FLOW_ATTENDANCE,
+                AttendanceRecord::STATE_PUNCHED_IN,
+                self::ESS_USER,
+                WorkflowStateMachine::ATTENDANCE_ACTION_EDIT_PUNCH_IN_TIME,
+                AttendanceRecord::STATE_PUNCHED_IN
+            );
+            $this->getAccessFlowStateMachineService()->deleteWorkflowStateMachineRecord(
+                WorkflowStateMachine::FLOW_ATTENDANCE,
+                AttendanceRecord::STATE_PUNCHED_OUT,
+                self::ESS_USER,
+                WorkflowStateMachine::ATTENDANCE_ACTION_EDIT_PUNCH_OUT_TIME,
+                AttendanceRecord::STATE_PUNCHED_OUT
+            );
+            $this->getAccessFlowStateMachineService()->deleteWorkflowStateMachineRecord(
+                WorkflowStateMachine::FLOW_ATTENDANCE,
+                AttendanceRecord::STATE_PUNCHED_OUT,
+                self::ESS_USER,
+                WorkflowStateMachine::ATTENDANCE_ACTION_EDIT_PUNCH_IN_TIME,
+                AttendanceRecord::STATE_PUNCHED_OUT
+            );
+            $this->getAccessFlowStateMachineService()->deleteWorkflowStateMachineRecord(
+                WorkflowStateMachine::FLOW_ATTENDANCE,
+                AttendanceRecord::STATE_PUNCHED_IN,
+                self::ESS_USER,
+                WorkflowStateMachine::ATTENDANCE_ACTION_DELETE,
+                AttendanceRecord::STATE_NA
+            );
+            $this->getAccessFlowStateMachineService()->deleteWorkflowStateMachineRecord(
+                WorkflowStateMachine::FLOW_ATTENDANCE,
+                AttendanceRecord::STATE_PUNCHED_OUT,
+                self::ESS_USER,
+                WorkflowStateMachine::ATTENDANCE_ACTION_DELETE,
+                AttendanceRecord::STATE_NA
+            );
```

Comment:
Need to move this kind of blocks to `AttendanceService`, in API you can simply call like `$this->getAttendanceService()->disable<Config>`